packages/data-context/src/sources/WizardDataSource.ts

```diff
@@ -1,57 +1,96 @@
 import type { DataContext } from '..'
 import path from 'path'
 import resolve from 'resolve-from'
 
 const debug = Debug('cypress:data-context:wizard-data-source')
 
+interface PackageToInstall {
+  name: AllPackageNames
+  description: AllPackagesDescriptions
+  package: AllPackagePackages
+  installer: typeof DEPENDENCIES[number]['installer']
+}
+
 export class WizardDataSource {
   constructor (private ctx: DataContext) {}
 
-  async packagesToInstall (): Promise<Array<NexusGenObjects['WizardNpmPackage']> | null> {
+  async packagesToInstall (): Promise<PackageToInstall[] | null> {
     if (!this.chosenFramework || !this.chosenBundler) {
       return null
     }
 
-    const packages = [
-      {
-        name: this.chosenFramework.name as string,
-        description: PACKAGES_DESCRIPTIONS[this.chosenFramework.package],
-        package: this.chosenFramework.package,
-      },
-      {
-        name: this.chosenBundler.name as string,
-        description: PACKAGES_DESCRIPTIONS[this.chosenBundler.package],
-        package: this.chosenBundler.package as string,
-      },
-    ]
+    const packages: PackageToInstall[] = [...this.chosenFramework.packages]
+
+    // find the matching dev server
+    // vite -> @cypress/vite-dev-server
+    // webpack -> @cypress/webpack-dev-server
+    // only applicable for `library` - `template` like CRA, Next.js etc
+    // only support 1 dev server, which is included in `framework.dependencies`.
+    if (this.chosenFramework.family === 'library') {
+      const cypressDevServer = CYPRESS_DEV_SERVERS.find(
+        (devServer) => devServer.supports.some((bundler) => bundler.type === this.chosenBundler?.type),
+      )
+
+      if (!cypressDevServer) {
+        throw Error(`Could not find matching Cypress Dev Server for ${this.chosenBundler.type}`)
+      }
+
+      packages.push(cypressDevServer)
+    }
 
     const storybookInfo = await this.ctx.storybook.loadStorybookInfo()
     const { storybookDep } = this.chosenFramework
 
     if (storybookInfo && storybookDep) {
-      packages.push({
-        name: storybookDep,
-        description: PACKAGES_DESCRIPTIONS[storybookDep],
-        package: storybookDep,
-      })
+      packages.push(storybookDep)
     }
 
     return packages
   }
 
-  async resolvePackagesToInstall (): Promise<string[]> {
+  async installDependenciesCommand () {
+    const commands = {
+      'npm': 'npm install -D',
+      'pnpm': 'pnpm install -D',
+      'yarn': 'yarn add -D',
+    }
+
+    let depsToInstall = (await this.ctx.wizard.packagesToInstall() ?? [])
+
+    if (!depsToInstall?.length) {
+      return null
+    }
+
+    const deps = depsToInstall.map((pack) => `${pack.installer}`).join(' ')
+
+    return `${commands[this.ctx.coreData.packageManager ?? 'npm']} ${deps}`
+  }
+
+  async installedPackages (): Promise<string[]> {
+    if (this.ctx.coreData.wizard.__fakeInstalledPackagesForTesting) {
+      return this.ctx.coreData.wizard.__fakeInstalledPackagesForTesting
```

Comment:
This is so we don't actually need to run a `npm install` during E2E tests; we simulate that by setting this to pretend like we ran `npm install`. That's what `__fakeInstalledPackagesForTesting` is for.